doc/lib/provisioning.txt

```diff
@@ -47,25 +47,34 @@ In addition to the components above, the Provisioning library also depends on C
 
-@section HighLevelOperation_Design High Level Provisioning Operation Design
+@section HighLevelOperation_Design High Level Design of a Provisioning Operation
+@brief The following shows a high-level workflow of each library operation. The library uses the MQTT stack for communicating with AWS IoT and the serializer stack for serializing and de-serializing request and responses, respectively, from the server.  
 @image html provisioning_operation_highlevel_design.png width=100%
 
-@section DetailedOperation_Design Detailed Provisioning Operation Workflow
+@section DetailedOperation_Design Detailed Workflow Diagram of a Provisioning Operation
+@brief The following shows the detailed workflow of each Provisioning operation. The Provisioning library architecture consists of 3 layers: API, serializer and parser, all of which are used for each Provisioning operation. The library uses an internal global state memory, separate for each operation type, that is shared between the Provisioning API and MQTT subscription callback layers. The global state represents the state of the operation and provides a synchronization mechanism between the Provisioning API and MQTT subscription callback layers.
 @image html provisioning_operation_detailed_design.png width=100%
 
 */
 
 /**
 @page provisioning_demo Demo
-@brief Demonstrates the usage of the Provisioning library.
-
-This demo provides a simple example on using the Provisioning library and working with the AWS IoT Fleet Provisioning feature.
-
-The provisioning demo establishes an MQTT connection and makes two API calls to the AWS IoT Core service through the @ref provisioning_function_createkeysandcertificate and @ref provisioning_function_registerthing functions. It registers callbacks with both the API functions for printing server responses. Also, the callback registered with the @ref provisioning_function_createkeysandcertificate function copies the certificate ID and certificate ownership token from the server response in separate buffers, which are then passed by the demo application in the call to the @ref provisioning_function_registerthing function for provisioning.
-
-@image html provisioning_demo.png "Provisioning Demo Workflow" width=80%
+@brief The demos demonstrate the usage of the Provisioning library for 2 different Provisioning workflows:
+<ul>
+<li> Provisioning when certificate is created by AWS IoT with a Certificate-Signing Request </li>
+<li> Provisioning when certificate and keys are by AWS IoT </li>
+</ul>
+This demos represent simple examples on using Provisioning library and working with the AWS IoT Fleet Provisioning feature.
+
+Both the demos demo establish an MQTT connection and make two API calls to the AWS IoT Core service for demonstrating the specific Provisioning workflow.
```

Comment:
I would recommend a rewording of this sentence. 
The words demo, demos and demonstrating doesn't really flow well in my opinion. 

One other correction is 'a MQTT connection'

Reply:
That's correct :p
Have re-worded the sentence.

Reply:
It would be 'an MQTT connection' as the pronunciation of MQTT starts with `em..` (i.e. starts with a vowel) 😃

Reply:
Ok. Just realized that. Thanks